pkg/validationfile/loader.go

```diff
@@ -121,6 +124,7 @@ func PopulateFromFilesContents(ctx context.Context, ds datastore.Datastore, file
 			ts, err := typesystem.NewNamespaceTypeSystem(objectDef,
 				typesystem.ResolverForDatastoreReader(rwt).WithPredefinedElements(typesystem.PredefinedElements{
 					Namespaces: objectDefs,
+					Caveats:    caveatDefs,
```

Comment:
I take it this was the root of it?

Reply:
Likely, yes, although I wasn't able to fully reproduce